xds/internal/resolver/serviceconfig.go

```diff
@@ -324,3 +370,33 @@ func (il *interceptorList) NewStream(ctx context.Context, ri iresolver.RPCInfo,
 	}
 	return newStream(ctx, func() {})
 }
+
+type clusterKey struct{}
+
+func GetPickedCluster(ctx context.Context) string {
+	return ctx.Value(clusterKey{}).(clusterInfoForLB).clusterName
+}
+
+func GetRequestHash(ctx context.Context) uint64 {
+	return ctx.Value(clusterKey{}).(clusterInfoForLB).requestHash
+}
+
+// GetPickedClusterForTesting returns the cluster in the context; to be used
+// for testing only.
+func GetPickedClusterForTesting(ctx context.Context) string {
+	return GetPickedCluster(ctx)
+}
+
+type clusterInfoForLB struct {
+	clusterName string
+	requestHash uint64
+}
+
+// SetPickedClusterAndRequestHash adds the selected cluster to the context for the
+// xds_cluster_manager LB policy to pick.
+func SetPickedClusterAndRequestHash(ctx context.Context, cluster string, requestHash uint64) context.Context {
```

Comment:
This actually can be unexported

Then you will need a `SetPickedClusterAndRequestHashForTesting()`.

Just like the old 
`getPickedCluster` and `GetPickedClusterForTesting` before this change.

Reply:
Changed.